README.md

```diff
@@ -1,244 +1,103 @@
-The environment variables for the lakeFS client arguments are the names of the constructor arguments prefixed with `LAKEFS_`:
-
-```python
-import os
-from lakefs_spec import LakeFSFileSystem
-
-os.environ["LAKEFS_HOST"] = "localhost:8000"
-os.environ["LAKEFS_USERNAME"] = "username"
-os.environ["LAKEFS_PASSWORD"] = "password"
-
-fs = LakeFSFileSystem()
-```
-
-To initialize the lakeFS file system from a `lakectl` YAML configuration file, you can specify the `configfile` argument.
-
-```python
-from lakefs_spec import LakeFSFileSystem
+REPO, BRANCH = "repo", "main"
 
-# No argument means the default config (~/.lakectl.yaml) will be used.
-fs = LakeFSFileSystem(configfile="path/to/my/lakectl.yaml")
-```
+# Prepare example local data
+local_path = Path("demo.txt")
+local_path.write_text("Hello, lakeFS!")
 
-⚠️ To be able to read settings from a YAML configuration file, `pyyaml` has to be installed, for example using `pip`:
+# Upload to lakeFS and create a commit
+fs = LakeFSFileSystem()  # will auto-discover config from ~/.lakectl.yaml
+repo_path = f"{REPO}/{BRANCH}/{local_path.name}"
+fs.put(str(local_path), repo_path)
+commit(fs.client, REPO, BRANCH, "Add demo data")
 
-```shell
-pip install --upgrade pyyaml
+# Read back committed file
+f = fs.open(repo_path, "rt")
+print(f.readline())  # "Hello, lakeFS!"
 ```
 
-### A note on mixing environment variables and `lakectl` configuration files
-
-lakeFS file system instances are cached, and existing lakeFS instances are reused from an instance cache when requested.
-
-For implicit initialization from environment variables and configuration files as described above, this means that whichever initialization method is used first populates the cache -
-thus, when using the other method, a cache hit happens and no new instance is created. This can lead to surprising misconfigurations:
-
-```python
-import os
-from lakefs_spec import LakeFSFileSystem
-
-# set envvars
-os.environ["LAKEFS_HOST"] = "localhost:8000"
-os.environ["LAKEFS_USERNAME"] = "username"
-os.environ["LAKEFS_PASSWORD"] = "password"
-
-# creates a cache entry for the bare instance
-fs = LakeFSFileSystem()
-
-# ~/.lakectl.yaml
-#  server:
-#    endpoint_url: http://example-host
+### High-level: Via third-party libraries
 
-# this time, try to read in the default lakectl config, with http://example-host set as host.
-fs = LakeFSFileSystem()
-print(fs.client._api.configuration.host) # <- prints localhost:8000!
-```
-
-The best way to avoid this is to commit to only using either environment variables or `lakectl` configuration files.
-If you do have to mix both methods, you can clear the instance cache like so:
+A variety of widely-used data science tools are building on `fsspec` to access remote storage resources and can thus work with lakeFS data lakes directly through `lakefs-spec`.
```

Comment:
Should we add a link to, or list of tools using `fsspec`. Eg. [this one](https://filesystem-spec.readthedocs.io/en/latest/).

Interesting tools using it might be Pandas, DuckDB, dask, kedro, Pyarrow, dvc, Intake.

Reply:
Done in [35ff4a6](https://github.com/appliedAI-Initiative/lakefs-spec/commit/35ff4a6d0ec34917ee03a911c3fe5a67d435a1d3)